### ISSUE TITLE
[Bug]: Unpaper Not Found: "Warning: using insecure memory!"

### ISSUE THREAD

**Issue by vfilby** [ocrmypdf/OCRmyPDF#1409]
### What were you trying to do?

TL;DR; Unpaper emits a warning to STDERR that interferes with the version matching regular expression in `get_version` because STDERR is piped to STDOUT in the subprocess execution.  Is there a reason why STDERR is piped to STDOUT or can this be fixed?

More detailed investigation below for search/interest purposes.

---
I have a docker image that makes use of ocrmypdf and when I was trying to setup the container on a different host I saw failures related to unpaper not being found despite it being on the host and in the PATH.

```
[05:16:45] DEBUG [ocrmypdf.subprocess.run:53] Running: ['unpaper', '--version']
[05:16:45] ERROR [ocrmypdf.subprocess._error_missing_program:262] The program 'unpaper' could not be executed or was not found on your system PATH.  This program is required when you use the ['--clean, --clean-final'] arguments.  You could try omitting these arguments, or install the package.
```

Running unpaper from the commandline seems to work as expected and it's in `/usr/bin` alongside other dependencies that were running fine.  

Trying to mimic the calls from unpaper directly I ran `get_version` for unpaper: 

```
>>> get_version('unpaper')
Traceback (most recent call last):
  File "/.venv/lib/python3.9/site-packages/ocrmypdf/subprocess.py", line 147, in get_version
    version = re.match(regex, output.strip()).group(1)
AttributeError: 'NoneType' object has no attribute 'group'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.venv/lib/python3.9/site-packages/ocrmypdf/subprocess.py", line 149, in get_version
    raise MissingDependencyError(
ocrmypdf.exceptions.MissingDependencyError: The program 'unpaper' did not report its version. Message was:
Warning: using insecure memory!
6.1
```

Digging deeper it looks like it's related to a warning that  unpaper emits to STDERR, `Warning: using insecure memory!`. I disregarded this when I was running unpaper at the commandline but this appears to cause problems with ocrmypdf because of the way the subprocess is executed. 

Specifically, stderr is piped into STDOUT and the warning is causing the version matching regular expression to fail.

```
        proc = run(
            args_prog,
            close_fds=True,
            text=True,
            stdout=PIPE,
            stderr=STDOUT,
            check=True,
            env=env,
        )
```

I don't know why unpaper is throwing the warning to STDERR, but that shouldn't matter to ocrmypdf since it /should/ be reading STDOUT.  Is there a reason why it pipes both together?  

### Where are you installing/running from?

Docker container

### OCRmyPDF version

11.3.3

### What operating system are you working on?

Linux

### Operating system details and version

Note that I am using a customer docker container that has worked without problem on a different host.

### Simple sanity checks

- [ ] Operating system is currently supported by its vendor (not end of life)
- [X] Python version is compatible with OCRmyPDF
- [ ] This issue is not about a specific input file

### Relevant log output

```plain text
[05:16:45] DEBUG [ocrmypdf.subprocess.run:53] Running: ['unpaper', '--version']
[05:16:45] ERROR [ocrmypdf.subprocess._error_missing_program:262]
The program 'unpaper' could not be executed or was not found on your
system PATH.  This program is required when you use the
['--clean, --clean-final'] arguments.  You could try omitting these arguments, or install
the package.

[05:16:45] INFO [ocrmypdf.subprocess._error_trailer:253]
On systems with the aptitude package manager (Debian, Ubuntu), try these
commands:
    sudo apt-get update
    sudo apt-get install unpaper
```

**Comment by v3nus-py** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by jbarlow83** [comment]
This issue concerns an old version of ocrmypdf so regrettably I cannot investigate unless it is reproducible in a current version. Since this is docker you should be able to upgrade easily.

stdout/stderr were combined because (at least historically) unpaper didn't seem to follow any clear rules about whether it wrote to stdout or stderr. Perhaps

**Comment by vfilby** [comment]
I updated the base of the docker image so that it pulls a more recent version of ocrmypdf.

```
root# ocrmypdf --version
16.5.0
```

The error message in the logs is:

```
19:54:21] ERROR [ocrmypdf.subprocess._error_missing_program:275]
The program 'unpaper' could not be executed or was not found on your
system PATH.  This program is required when you use the
--clean, --clean-final arguments.  You could try omitting these arguments, or install
the package.

[19:54:21] INFO [ocrmypdf.subprocess._error_trailer:263]
On systems with the aptitude package manager (Debian, Ubuntu), try these
commands:
    sudo apt update
    sudo apt install unpaper

On RPM-based systems (Red Hat, Fedora), try this command:
    sudo dnf install unpaper

[19:54:21] ERROR [root.create_searchable_pdf:198] The program 'unpaper' did not report its version. Message was:
Warning: using insecure memory!
7.0.0
```

I have a work around for this, I am wrapping the unpaper binary and sending STDERR to /dev/null....  not ideal but it gets me by the version check.
